lib/ansible/plugins/loader.py

```diff
@@ -48,6 +48,7 @@ def get_all_plugin_loaders():
 def add_all_plugin_dirs(path):
     ''' add any existing plugin dirs in the path provided '''
     b_path = to_bytes(path, errors='surrogate_or_strict')
```

Comment:
```suggestion
     b_path = os.path.expanduser(to_bytes(path, errors='surrogate_or_strict'))
```

Reply:
Could you please add a changelog entry for this change? Also, test for this change would be good to have.